crypto/thread/arch/thread_none.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019-2020 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2002, Oracle and/or its affiliates. All rights reserved
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/configuration.h>
+#include <openssl/e_os2.h>
+
+#if defined(_WIN32)
+# include <windows.h>
+#endif
+
+#if !defined(OPENSSL_THREADS) || ( \
+    !defined(OPENSSL_SYS_UNIX) && !defined(OPENSSL_SYS_WINDOWS) ) || (\
+    defined(_WIN32_WINNT) && _WIN32_WINNT < 0x0600 )
+
+# include <internal/thread.h>
+
+int CRYPTO_THREAD_native_create(CRYPTO_THREAD thread)
+{
+    (void)thread;
+    return 0;
+}
+
+int CRYPTO_THREAD_native_join(CRYPTO_THREAD thread, CRYPTO_THREAD_RETVAL *retval)
+{
+    (void)thread;
+    (void)retval;
```

Comment:
On the other hand they are harmless.

Reply:
>> int crypto_thread_native_spawn(CRYPTO_THREAD *thread)
>> {
>>    (void)thread;
>>    return 0;
>> }
>> </snip>

> These lines should not be necessary (we don't do this anywhere else for unused args).

@mattcaswell would you prefer to have them removed or is it acceptable to keep them in (in which case, could you please mark this as resolved)? Thanks!

Reply:
TBH, I'd really like to see them removed.

Reply:
Instead of the cast to `(void)`, prefix the argument with `ossl_unused`:
```c
int crypto_thread_native_join(ossl_unused CRYPTO_THREAD *thread, ossl_unused CRYPTO_THREAD_RETVAL *retval)
```